### PR TITLE
feat: Ruby 3.3.1

### PR DESCRIPTION
**Pull request by kgilpin**
Try adding 3.3.1 to the test matrix

### CODE REVIEW

**Comment by kgilpin** [comment]
Possibly helpful - https://rubyreferences.github.io/rubychanges/3.3.html

File: ext/appmap/appmap.c [hunk 1]
```diff
@@ -17,7 +18,13 @@ static VALUE
 singleton_method_owner_name(VALUE klass, VALUE method)
 {
   VALUE owner = rb_funcall(method, rb_intern("owner"), 0);
+
+#if RUBY_API_VERSION_CODE < 30200
```
Comment (kgilpin) [inline ext/appmap/appmap.c:22]: Whoa

Comment (apotterri) [reply ext/appmap/appmap.c:22]: It was pretty grotty before. The change they made in 3.2 allows it to be slightly less so.

**Review by apotterri — APPROVED** [review]

**Pull request merged**